redis-memory-server fails to compile Redis native modules on json-rpc-relay runner (missing cmake, cargo, pkg-config, libssl-dev)

redis-memory-server fails to compile Redis native modules on json-rpc-relay runner (missing cmake, cargo, pkg-config, libssl-dev)

## Root cause

The `Install NodeJS Dependencies (json-rpc-relay)` step (`npm ci` in `regression-tests/json-rpc-relay`) runs the `redis-memory-server` postinstall. That package ships **no prebuilt binaries** — it downloads the Redis **source** and compiles it with `make`. The relay project does not pin a version, so it uses the package default `"stable"`, which resolves to the **rolling** tarball:

```
https://download.redis.io/redis-stable.tar.gz
```

(URL construction: [`redis-memory-server/src/util/RedisBinaryDownloadUrl.ts`](https://github.com/mhassan1/redis-memory-server/blob/main/src/util/RedisBinaryDownloadUrl.ts) — `version === "stable"` → `${mirror}/redis-stable.tar.gz`.)

That `stable` pointer **rolled forward to Redis 8.10.0 on 2026-07-29** (`https://download.redis.io/redis-stable/src/version.h` currently reads `8.10.0`). Redis 8's `make` additionally builds the bundled modules (RedisBloom / RediSearch / RedisJSON / RedisTimeSeries), and those module builds require a toolchain that is **not** present on the `hl-cn-rpc-relay-lin-lg` runner (`pkg-config`, OpenSSL headers/`libssl-dev`, and `cmake`/`cargo` for others).

## This is not a container/image regression

Worth stating explicitly, since the title could be read as "the runner lost its build utils": **the runner image did not change and nothing was removed from it.** What changed is *which source gets compiled inside it*.

- Before 2026-07-29, `stable` was on the Redis **7.x** line. Redis 7's `make` builds only `redis-server` — no modules, TLS off by default — so it compiles with the bare `gcc`/`make` the image has always had. `pkg-config`/`libssl-dev` were never needed.
- After the pointer moved to Redis **8.10.0**, `make` also builds the modules, which *do* need that toolchain.

The image was always missing those packages; it just never mattered until the compiled source began requiring them. The log confirms this — the `pkg-config: not found` lines during the core build are non-fatal warnings; the actual fatal error is inside a **module** directory that only exists in Redis 8:

```
npm error .../node_modules/.cache/redis-memory-server/redis-binaries/stable/extracted/
  modules/redistimeseries/src/deps/LibMR/deps/hiredis/ssl.c:55:10:
  fatal error: openssl/ssl.h: No such file or directory
npm error make: *** [Makefile:109: build] Error 1
npm error command failed: sh -c node ./scripts/postinstall
##[error]Process completed with exit code 1.
```

## Evidence / regression window

The JSON-RPC Relay panel's own run history brackets the break exactly to the Redis 8.10.0 release (2026-07-29). Runs where the panel executed:

| Time (UTC) | Relay panel | Run |
|---|---|---|
| 2026-07-29 00:43 | ✅ success | 30411963011 |
| — Redis 8.10.0 released 2026-07-29 — | | |
| 2026-07-29 21:22 | ❌ failure | 30492085190 |
| 2026-07-30 09:53 | ❌ failure | 30532523498 |
| 2026-07-30 15:41 | ❌ failure | [30557917505](https://github.com/hiero-ledger/hiero-consensus-node/actions/runs/30557917505/job/90926947948) |

Every pre-8.10.0 run passed on the same bare image; every run after the pointer moved has failed at the same compile step. (The green cron runs in between are ones where this panel was gated off, not passes.)

Note: this XTS panel runs with `REDIS_ENABLED: false`, so the compiled Redis binary is never actually started — it only has to *install*.

## Fix applied

Pin the Redis version `redis-memory-server` compiles, so the install stops tracking the moving `stable` pointer, in `.github/workflows/818-call-json-rpc-relay-regression.yaml` (the `Install NodeJS Dependencies (json-rpc-relay)` step):

```yaml
      - name: Install NodeJS Dependencies (json-rpc-relay)
        env:
          REDISMS_VERSION: "7.4.10"
        run: npm ci
        working-directory: regression-tests/json-rpc-relay
```

`7.4.10` is the latest patch of the last pre-Redis-8 line — module-free, verified present at `https://download.redis.io/releases/redis-7.4.10.tar.gz` (HTTP 200), and it compiles with the runner's existing toolchain, restoring the previously-working behavior. No apt/toolchain step is needed (the passing runs didn't have one).

### Alternatives considered

- **Install the full Redis 8 build toolchain** on the runner (`pkg-config`, `libssl-dev`, `build-essential`, `cmake`, `python3`, Rust) and let it compile `stable`. Keeps upstream default, but slow (multi-minute compile) and fragile, and leaves us tracking a moving pointer.
- **Use a distro `redis-server`** via `REDISMS_SYSTEM_BINARY` (`apt-get install -y redis-server`). Robust and fast, but bypasses the compile entirely.

Pinning was chosen because the root cause is the *unpinned, moving* `stable` pointer; pinning makes the install deterministic and immune to the next Redis release.

## Verification

1. Trigger the JSON-RPC Relay panel (via the cron suite or the manual dry-run `001-user-dry-run-extended-test-suite.yaml`).
2. In the install step, confirm the log shows `Downloading Redis 7.4.10` (not `stable`), no `pkg-config`/`openssl/ssl.h`/`make Error`, and `npm ci` exits 0.
3. Confirm the job proceeds past install into Install Solo → Configure and run solo → Start json-rpc-relay-test-client.

--- a/.github/workflows/818-call-json-rpc-relay-regression.yaml
+++ b/.github/workflows/818-call-json-rpc-relay-regression.yaml
@@ -119,6 +119,13 @@ jobs:
 
       # Set up the npm dependencies and cache on the json-rpc-relay repository
       - name: Install NodeJS Dependencies (json-rpc-relay)
+        env:
+          # Pin the Redis binary that redis-memory-server compiles in its postinstall. It otherwise
+          # tracks the rolling "stable" tarball (download.redis.io/redis-stable.tar.gz), which moved to
+          # Redis 8.10.0 on 2026-07-29 and now builds bundled modules needing a toolchain
+          # (pkg-config/libssl-dev/cmake/rust) absent on this runner. 7.4.x is module-free and compiles
+          # with the runner's existing toolchain (the last pre-8 line that worked here).
+          REDISMS_VERSION: "7.4.10"
         run: npm ci
         working-directory: regression-tests/json-rpc-relay
 

